### PR TITLE
Prioritizes syntax at the current cursor position for determining the file type

### PR DESCRIPTION
**Pull request by jpsouzasilva**
### Motivation

A `.vue` file usually has three blocks: 
- template (HTML, ...)
- script (JS, Typescript, ...)
- style (LESS, SASS, CSS, ...)

All inside of the same file. emmet-vim was not correctly identifying the syntax provided for each of those sections and always expanding the abbreviation as HTML instead.

### Goal

Some plugins are dedicated to providing the correct syntax for each code block inside a file, thus are more accurate for their specific filetype targets (e.g. https://github.com/posva/vim-vue). 

screenshot from 2018-08-25 15-39-35

screenshot from 2018-08-25 15-42-30

The main goal here is to **rely on them first** and then fallback to the internal logic if it doesn't work. In practice, reorder the filetype identification sequence in order to prioritize the syntax provided for the current cursor's block.

### CODE REVIEW

**Comment by mattn** [comment]
Thank you. test seems to be fail

**Comment by jpsouzasilva** [comment]
@mattn could you tell me how to run the tests locally? so I don't have to keep pushing to see how it goes through the build

**Comment by jpsouzasilva** [comment]
@mattn running the tests locally with **make test**, all of them pass without issues. Do you think it could be hanging because `emmet#util#getcurpos()` is now at the top of the function?

File: autoload/emmet.vim [hunk 1]
```diff
@@ -376,10 +376,21 @@ endfunction
...
+    return &filetype
+  endif 
+
+  let pos = emmet#util#getcurpos()
+  let type = synIDattr(synID(pos[1], pos[2], 1), 'name')
+  if type =~? '^css\w'
+    let type = 'css'
+  elseif type =~? '^html\w'
+    let type = 'html'
+  elseif type =~? '^js\w' || '^javascript'
```
Comment (jpsouzasilva) [inline autoload/emmet.vim]: @mattn using case-insensitive matching now that works for both "HTML" and "html", "CSS" and "css", etc... : - ]

**Comment by jpsouzasilva** [comment]
@mattn Seems like it's passing now \o/

**Comment by jpsouzasilva** [comment]
@mattn The build is passing now. I've changed the code back to matching on both "js" and "javascript" insensitively (ignore-case).

File: autoload/emmet.vim [hunk 2]
```diff
@@ -377,38 +377,38 @@ endfunction
...
   let pos = emmet#util#getcurpos()
   let type = synIDattr(synID(pos[1], pos[2], 1), 'name')
-  if type =~# '^css\w'
+  if type =~? '^css\w'
     let type = 'css'
-  elseif type =~# '^html\w'
+  elseif type =~? '^html\w'
     let type = 'html'
-  elseif type =~# '^js\w'
+  elseif type =~? '^js\w' || '^javascript'
```
Comment (mattn) [inline autoload/emmet.vim]:
```
elseif type =~? '^js\w' || type =~? '^javascript'
```

**Comment by jpsouzasilva** [comment]
@mattn just after you pointed out the matching for javascript, I've added a clause tag for JSX just above since they share the same prefix. I believe for the well-known types it's better now. Is something missing?

File: autoload/emmet.vim [hunk 3]
```diff
@@ -523,7 +520,7 @@ function! emmet#expandAbbr(mode, abbr) range abort
   if len(filters) ==# 0
     let filters = ['html']
   endif
-
+
```
Comment (mattn) [inline autoload/emmet.vim]: nit pick. sorry. please remove extra spaces.

**Comment by mattn** [comment]
Thank you

**Pull request merged**